web/src/markdown.js

```diff
@@ -388,7 +388,25 @@ function make_emoji_span(codepoint, title, alt_text) {
 }
 
 function handleUnicodeEmoji({unicode_emoji, get_emoji_name}) {
-    const codepoint = unicode_emoji.codePointAt(0).toString(16);
+    const emoji_codepoints = [];
+    for (let i = 0; i < unicode_emoji.length; i += 1) {
+        const char_codepoint = unicode_emoji.codePointAt(i).toString(16);
+        // TODO: I don't know why the ones starting with d need to be removed
```

Comment:
@andersk do you know what's up with that?

Reply:
You have discovered UTF-16 surrogate pairs. In JavaScript, indexing into strings gives you UTF-16 code units, not Unicode code points.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#utf-16_characters_unicode_code_points_and_grapheme_clusters

Use iteration instead: `[...unicode_emoji].map(char => char.codePointAt(0).toString(16).padStart(4, '0'))`.

Reply:
Thanks! Did that.